client/app/pages/queries/query.html

```diff
@@ -74,6 +74,22 @@
             </a>
         </span>
       </h3>
+      <div class="compare-query-version">
+        <span class="zmdi zmdi-archive"></span>
+        <span class="text-muted">Version</span>
+        <strong>{{query.version}}</strong>
+        &gt;
+        <span>Compare to</span>
+        <select id="version-choice">
+          <option ng-repeat="version in versions" value="{{version.object_version}}">
+            {{version.object_version}}
+          </option>
+        </select>
+        <button class="btn btn-s btn-default" ng-click="compareQueryVersion()">
+          <span class="zmdi zmdi-play"></span>
+          Compare
+        </button>
+      </div>
```

Comment:
Next time, it will be great if you could include a screenshot of all the UI changes. But based on the markup -- I think it will take too much space.

How about linking the `updated at` timestamp to a UI similar to what Dropbox does in their Paper product:

![image](https://cloud.githubusercontent.com/assets/71468/23091839/ffdf7ca8-f5c7-11e6-8d7c-28572793d0a0.png)

Basically they show a list of all the recent changes. It's both efficient on the screen real estate, but also efficient in reviewing the changes, as you don't have to pick each individual version.

You just need to change the changes list API to return the full change set and then we save on multiple API calls. Another thing to consider is that not all changes are changes to the query text, so need to filter those out (can be done in the client side).